File: src/pandora/service/article.clj
(ns pandora.service.article
  (:require [pandora.gateway.mediawiki.mercury :as mercury]
            [pandora.domain.article :as article]))

(defn article!
  [wikia article-name]
  (let [resp (mercury/get-article! wikia article-name)
        {body :body success :success} (mercury/resolve-request resp)]
    (when body
      (let [name (mercury/title body)
            headline (mercury/abstract body)
            articleBody (mercury/content body)
            id (mercury/id body)]
        (article/->Article name id headline articleBody (article/url name))))))

File: service/facade/src/main/java/com/wikia/pandora/domain/Article.java
package com.wikia.pandora.domain;


public class Article {

  private final Integer id;
  private final String title;
  private final Integer ns;

  public Article(Integer id, String title, Integer ns) {
    this.id = id;
    this.title = title;
    this.ns = ns;
  }


  public Integer getId() {
    return id;
  }

  public String getTitle() {
    return title;
  }


  public Integer getNs() {
    return ns;
  }
}
